gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersions.java

```diff
@@ -98,38 +109,9 @@ private static boolean isLibrary(Project project, BaselineJavaVersionExtension p
                     project.getDisplayName());
             return false;
         }
-        ImmutableList<String> jarPublications = publishing.getPublications().stream()
-                .filter(pub -> isLibraryPublication(project, pub))
-                .map(Named::getName)
-                .collect(ImmutableList.toImmutableList());
-        if (jarPublications.isEmpty()) {
-            log.debug(
-                    "Project '{}' is considered a distribution because it does not publish jars",
-                    project.getDisplayName());
-            return false;
-        }
-        log.debug(
-                "Project '{}' is considered a library because it publishes jars: {}",
-                project.getDisplayName(),
-                jarPublications);
-        return true;
-    }
 
-    private static boolean isLibraryPublication(Project project, Publication publication) {
-        if (publication instanceof MavenPublication) {
-            MavenPublication mavenPublication = (MavenPublication) publication;
-            return mavenPublication.getArtifacts().stream().anyMatch(artifact -> "jar".equals(artifact.getExtension()));
-        }
-        if (publication instanceof IvyPublication) {
-            IvyPublication ivyPublication = (IvyPublication) publication;
-            return ivyPublication.getArtifacts().stream().anyMatch(artifact -> "jar".equals(artifact.getExtension()));
-        }
-        // Default to true for unknown publication types to avoid setting higher jvm targets than necessary
-        log.warn(
-                "Unknown publication '{}' of type '{}'. Assuming project {} is a library",
-                publication,
-                publication.getClass().getName(),
-                project.getName());
```

Comment:
There's an argument to be made for hiding this logic behind a gradle 8.6 version comparison to begin with, so we can experiment without impacting all gradle-baseline users in case something goes wrong. Not something that I'd want to keep long-term, but may make initial rollout a bit easier

Reply:
Nice idea, I think I'll do this. We can then just check all the Gradle upgrade excavators made after a certain date are merging.

Reply:
Although I think the behaviour will still be different unless I switch all the old logic and all the new logic on the gradle version (and not have the new plugins)

Reply:
Kept the old behaviour for now, need a +1.